Improve fling

radderz opened the pull request:
This finishes fling logic, works well on more sensitive screens now.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Mapsui/Mapsui?pullRequest=663) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/Mapsui/Mapsui?pullRequest=663) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: radderz<br/>:x: Mark Radcliffe<br/><hr/>**Mark Radcliffe** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Mapsui/Mapsui?pullRequest=663) it.</sub>

radderz commented:
@pauldendulk - I don't get the CLA thing, not sure how the commits are unrelated to my account or how to change/fix that?

radderz commented:
> @pauldendulk - I don't get the CLA thing, not sure how the commits are unrelated to my account or how to change/fix that?

Nevermind, I saw the small text, had to add my other email address to the account

charlenni commented:
Thank you @radderz for looking into this.

One small note: if I remember it correct, @pauldendulk moves back to SkiaSharp 1.60.0, because Rotation on 1.68.0 was to bad. But I could not found the message for this.

radderz commented:
@pauldendulk - I have updated in this branch to the latest Xamarin.Forms and updated to the latest SkiaSharp, is there a compelling reason to not update either of these? Having the older version of skiasharp limits other libraries but if it doesn't work well I'll revert it. 

Was it the tearing issue?

radderz commented:
> Thank you @radderz for looking into this.
> 
> One small note: if I remember it correct, @pauldendulk moves back to SkiaSharp 1.60.0, because Rotation on 1.68.0 was to bad. But I could not found the message for this.

right, I have other things using the newer version of skiasharp, but I can revert if needed. Do we know why there was an issue? I remember a tearing conversation at one point?

charlenni commented:
Found it. Was #643.

radderz commented:
@charlenni @pauldendulk 

I have reverted it back to 1.60.3

radderz commented:
@charlenni @pauldendulk , after reverting I still get the tearing in android on slower phones so its not the newer version of Skia causing it

radderz commented:
@pauldendulk can we merge this one in and deploy an update nuget? we have been running this branch in production for a while and fixes some pretty big issues with the forms implementation.

radderz commented:
@charlenni are you able to review/complete this pull request?

charlenni commented:
Sorry, but it seems, that I could not review this PR. And I could definitly not create a new nuget package.

radderz commented:
@charlenni - You appear to be a contributor, so I had thought you could do code reviews, but that's fine if you can't. Paul is obviously a bit busy at the moment as he has been off this project for a while. I can't review my own pull request unfortunately.

charlenni commented:
It seems, that I could review other PRs, but not this. Perhaps it is to old? Could you try to close it and add it new? And the problem with the nuget package still exists.

radderz commented:
> It seems, that I could review other PRs, but not this. Perhaps it is to old? Could you try to close it and add it new? And the problem with the nuget package still exists.

Can you try now? I may have requested Paul to do a review a while back, I have removed that request.

charlenni commented:
It seems, that I'm not enough for this work :( Sorry.

But if you could approve PRs, than you should have write permission. Perhaps I could insert your changes into a new branch and make a new PR, so that you could approve it?

charlenni commented:
I found [this](https://twitter.com/mattleibow/status/1160691570603167744) message on Twitter, which leads to [this PR](https://github.com/mono/SkiaSharp.Extended/tree/dev/controls/SkiaSharp.Extended.Controls). This could be interessting.

pauldendulk commented:
@radderz @charlenni I will slowly start to pick things up after a long break. The original PR is from May. This is still relevant?

charlenni commented:
Overall I would say yes. 

Mark and I use different approaches. Mark use a overall velocity of the last 200 ms, where I use the last few velocities. My approachs is slower, because of a linear regresion calculation.

I assume, that we should test it. Only the feeling while using it could show, if Marks approach feels good.

radderz commented:
@pauldendulk are you going to complete the pull request any time soon? It's starting to get conflicts now and this branch has a pretty big fix for anyone using fling.

pauldendulk commented:
> @pauldendulk are you going to complete the pull request any time soon? It's starting to get conflicts now and this branch has a pretty big fix for anyone using fling.

In the coming weeks. Sorry for the day.

pauldendulk commented on Mapsui.UI.Forms/Utils/FlingTracker.cs:
```diff
@@ -26,9 +26,10 @@ public void AddEvent(long id, Geometries.Point location, long ticks)
             
             // Check, if we at the end of array
-            if (events[id].Count > maxSize)
+            if (events[id].Count > 2)
```
There is a check on > 2 in the if but it is not in the while loop. If there really should be a minimum length of 2 it would be better to check that in the loop. There is a (very small) change the queue length will change after the > 2 check and before (or during) the while. Also it makes it a bit logical to read for me.
